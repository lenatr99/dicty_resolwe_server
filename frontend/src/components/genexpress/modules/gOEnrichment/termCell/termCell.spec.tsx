import React from 'react';
import { Mock, vi } from 'vitest';
import GOEnrichmentTermCell from './termCell';
import { customRender } from 'tests/test-utils';
import { generateGOEnrichmentRow } from 'tests/mock';
import { GOEnrichmentRow } from 'redux/models/internal';

describe('gOEnrichmentTermCell', () => {
    let goEnrichmentRow: GOEnrichmentRow;
    let asFragment: () => DocumentFragment;
    let mockedOnToggleCollapseClick: Mock;

    beforeEach(() => {
        mockedOnToggleCollapseClick = vi.fn();
        goEnrichmentRow = generateGOEnrichmentRow(1);
        goEnrichmentRow.term_name = 'Test term';
        goEnrichmentRow.depth = 3;

        ({ asFragment } = customRender(
            <GOEnrichmentTermCell
                data={goEnrichmentRow}
                onToggleCollapseClick={mockedOnToggleCollapseClick}
            />,
        ));
    });

    it('should render as default snapshot', () => {
        expect(asFragment()).toMatchSnapshot();
    });
});
