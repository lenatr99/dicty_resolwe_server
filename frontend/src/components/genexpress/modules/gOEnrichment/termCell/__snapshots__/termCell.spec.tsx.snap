// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`gOEnrichmentTermCell > should render as default snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu ciisdu"
  >
    <span
      class="sc-gJhJTp jOlInc"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root-kxJCyd dhFBup MuiSvgIcon-root MuiSvgIcon-fontSizeMedium sc-elDIKY cngcVL"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m7 10 5 5 5-5z"
      />
    </svg>
    <span>
      Test term
    </span>
  </div>
</DocumentFragment>
`;
